lib/actions-oclif/device/init.ts

```diff
@@ -0,0 +1,172 @@
+				`,
+		}),
+		drive: flags.string({
+			char: 'd',
+			description: stripIndent`
+				the drive to write the image to, eg. \`/dev/sdb\` or \`/dev/mmcblk0\`. \
+				Careful with this as you can erase your hard drive. \
+				Check \`balena util available-drives\` for available options.
+				`,
+		}),
+		config: flags.string({
+			description: 'path to the config JSON file, see `balena os build-config`',
+		}),
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { flags: options } = this.parse<FlagsDef, {}>(DeviceInitCmd);
+
+		// Imports
+		const { promisify } = await import('bluebird');
+		const rimraf = promisify(await import('rimraf'));
+		const tmp = await import('tmp');
+		const tmpNameAsync = promisify(tmp.tmpName);
+		tmp.setGracefulCleanup();
+		const balena = getBalenaSdk();
+		const patterns = await import('../../utils/patterns');
+		const Logger = await import('../../utils/logger');
+
+		const logger = Logger.getLogger();
+
+		// Consolidate application options
+		options.application = options.application || options.app;
+		delete options.app;
+
+		// Get application and
+		const application = await balena.models.application.get(
+			options['application'] || (await patterns.selectApplication()),
+		);
+
+		// Register new device
+		const deviceUuid = balena.models.device.generateUniqueKey();
+		console.info(`Registering to ${application.app_name}: ${deviceUuid}`);
+		await balena.models.device.register(application.id, deviceUuid);
+		const device = await balena.models.device.get(deviceUuid);
+
+		// Download OS, configure, and flash
+		const tmpPath = (await tmpNameAsync()) as string;
+		try {
+			logger.logDebug(`Downloading OS image...`);
+			await this.downloadOsImage(tmpPath, application.device_type, options);
+
+			logger.logDebug(`Configuring OS image...`);
+			await this.configureOsImage(tmpPath, device.uuid, options);
+
+			logger.logDebug(`Writing OS image...`);
+			await this.writeOsImage(tmpPath, application.device_type, options);
+		} catch (e) {
+			// Remove device in failed cases
+			try {
+				logger.logDebug(`Process failed, removing device ${device.uuid}`);
+				await balena.models.device.remove(device.uuid);
+			} catch (e) {
+				// Ignore removal failures, and throw original error
+			}
+			throw e;
+		} finally {
+			// Remove temp download
+			logger.logDebug(`Removing temporary OS image download...`);
+			await rimraf(tmpPath);
+		}
+
+		console.log('Done');
+		return device.uuid;
+	}
+
+	async downloadOsImage(path: string, deviceType: string, options: FlagsDef) {
+		const osVersion = options['os-version'] || 'default';
+		await runCommand(
+			`os download ${deviceType} --output '${path}' --version ${osVersion}`,
+		);
+	}
+
+	async configureOsImage(path: string, uuid: string, options: FlagsDef) {
+		let configureCommand = `os configure ${path} --device ${uuid}`;
+		if (options.config) {
+			configureCommand += ` --config ${options.config}`;
```

Comment:
It seems that the single quotes needed to be removed around the paths here (I think it must be due to oclif and capitano parsing differences).

Reply:
See my other comment in the `runCommand` function: I suggest using arrays all along:
```
let configureCommand = ['os', 'configure', path, '--device', uuid];
if (options.config) {
    configureCommand.push('--config', options.config);
```